api/v3/PaymentProcessor/GetRefundedAmount.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/**
+ * Action get refundable balance.
+ *
+ * @param array $params
+ *
+ * @return array
+ *   API result array.
+ *
+ * @throws \API_Exception
+ * @throws \CiviCRM_API3_Exception
+ * @throws \Civi\Payment\Exception\PaymentProcessorException
+ */
+function civicrm_api3_payment_processor_get_refunded_amount($params) {
+  /** @var \CRM_Core_Payment $processor */
+  $processor = Civi\Payment\System::singleton()->getById($params['payment_processor_id']);
+  $processor->setPaymentProcessor(civicrm_api3('PaymentProcessor', 'getsingle', ['id' => $params['payment_processor_id']]));
+  if (!$processor->supportsRefund()) {
+    throw new API_Exception('Payment Processor does not support refund');
+  }
+  if (!method_exists($processor, 'getRefundedAmountByChargeId')) {
+    throw new API_Exception('Payment Processor does not support refund amount');
```

Comment:
So, we are throwing the error if there is any implementation of `getRefundedAmountByChargeId` method. 

Are we still showing a refund links on a contribution list if payment processor does not implement this? You may want to update the logic on \Civi\Financeextras\Payment\Refund.

Reply:
Hmm, I think better to not throw error as PaymentProcessor has Refund API but not "RefundedAmout" Api may be we can update UI to reflect that, like Remove Refunded Amount / Available Amount in UI or display some warning. 

Let me know your views.